deepmd/utils/argcheck.py

```diff
@@ -767,19 +789,143 @@ def normalize_hybrid_list(hy_list):
+    single_fitting_net = "fitting_net" in data["model"].keys()
+    single_training_data = "training_data" in data["training"].keys()
+    single_valid_data = "validation_data" in data["training"].keys()
+    single_loss = "loss" in data.keys()
+    multi_fitting_net = "fitting_net_dict" in data["model"].keys()
+    multi_training_data = "data_dict" in data["training"].keys()
+    multi_loss = "loss_dict" in data.keys()
+    multi_fitting_weight = "fitting_weight" in data["training"].keys()
+    assert (single_fitting_net == single_training_data) and \
+           (multi_fitting_net == multi_training_data), \
+            "In single-task mode, 'model/fitting_net' and 'training/training_data' must be defined at the same time! " \
+            "While in multi-task mode, 'model/fitting_net_dict', 'training/data_dict' " \
+            "must be defined at the same time! Please check your input script. "
+    assert not (single_fitting_net and multi_fitting_net), \
+        "Single-task mode and multi-task mode can not be performed together. " \
+        "Please check your input script and choose just one format! "
+    assert single_fitting_net or multi_fitting_net, "Please define your fitting net and training data! "
+    if multi_fitting_net:
+        assert not single_valid_data, "In multi-task mode, 'training/validation_data' should not appear " \
+                                      "outside 'training/data_dict'! Please check your input script."
+        assert not single_loss, "In multi-task mode, please use 'model/loss_dict' in stead of 'model/loss'! "
+        assert "type_map" in data["model"], "In multi-task mode, 'model/type_map' must be defined! "
+        data["model"]["fitting_net_dict"] = normalize_fitting_net_dict(data["model"]["fitting_net_dict"])
+        data["training"]["data_dict"] = normalize_data_dict(data["training"]["data_dict"])
+        data["loss_dict"] = normalize_loss_dict(data["model"]["fitting_net_dict"].keys(),
+                                                data["loss_dict"]) if multi_loss else {}
+        fitting_weight = data["training"]["fitting_weight"] if multi_fitting_weight else None
+        data["training"]["fitting_weight"] = \
+            normalize_fitting_weight(data["model"]["fitting_net_dict"].keys(),
+                                     data["training"]["data_dict"].keys(),
+                                     fitting_weight=fitting_weight)
+    else:
+        assert not multi_loss, "In single-task mode, please use 'model/loss' in stead of 'model/loss_dict'! "
+    return data
+
+
+def normalize_fitting_net_dict(fitting_net_dict):
+    new_dict = {}
+    base = Argument("base", dict, [], [fitting_variant_type_args()], doc="")
+    for fitting_key_item in fitting_net_dict:
+        data = base.normalize_value(fitting_net_dict[fitting_key_item], trim_pattern="_*")
+        base.check_value(data, strict=True)
+        new_dict[fitting_key_item] = data
+    return new_dict
```

Comment:
This could be something implemented in dargs. Currently, we have `repeat=True` for `list`, but it is possible to make it also available for `dict` (`dict.values()`). ping @y1xiaoc

Reply:
I'm not sure if that is desirable. `repeat` is here to deal with the case where one may have indefinitely many homogeneous dicts in that list. For dict, to make the type checking work, one need to specify the keys in advance. Therefore, there should be a fixed set of keys although the corresponding values are the same type. In that case, one should just write a loop to generate the Argument. If the use case is that no matter what the key is, the values follows the same dict type, maybe one should reconsider reform it into a list of dict with the original key being one of the attributes.

Reply:
After discussion with @wanghan-iapcm , maybe it's not a suitable choice to make fitting_net_dict into a `list`: 
1. users may have to manully pair the fitting_net and its training datasystems;
2. or we must add an additional key in fitting_net to identify itself, which may have more conflicts than `dict`.

Reply:
It's seems that I didn't get the full context. As I mentioned if there are finite choices of keys then this can be directly written as a loop to generate the Argument. The only use case is there are indefinite choices of keys. Is that the case? Or maybe we can schedule a video call on this.